torch/csrc/distributed/rpc/rref_context.cpp

```diff
@@ -18,10 +18,14 @@ void confirmPendingUser(
   if (!futureMessage.hasError()) {
     auto rr = RemoteRet::fromMessage(futureMessage.constValue());
     TORCH_INTERNAL_ASSERT(rr->forkId() == expectedForkId);
+  } else {
+    // Handle errors, such as timeouts, by invoking the error handler on the
+    // rref.
+    auto rref_ptr = RRefContext::getInstance().getPendingUser(expectedForkId);
+    auto errorType = getRPCErrorType(futureMessage);
+    rref_ptr->handleError(errorType, futureMessage);
   }
   RRefContext::getInstance().delPendingUser(expectedForkId);
-  // Potentially propagate to the userRRef?
-  RRefContext::handleException(futureMessage);
 }
 
 c10::intrusive_ptr<RRef> finishCreatingOwnerRRef(
```

Comment:
Do we also need to check timeout error here, in case the `remote` is sent to self?